File: Library_project/Periodic.cpp
#include "Periodic.h"


Periodic::Periodic(const string& call_no,const string& title,const string& subjects,const string& author,const string& description,const string& publisher,const string& publishing_history,const string& series,const string& notes,const string& related_titles,const string& other_forms_of_title,const string& gov_doc_number):Media(call_no,title,subjects,notes),author(author), description(description), publisher(publisher),publishing_history(publishing_history),series(series),related_titles(related_titles),other_forms_of_title(other_forms_of_title),gov_doc_number(gov_doc_number)
{
}

Periodic::~Periodic(){}


/*
   compare_other finds if the input string exists in Description,Notes, series
   and related_titles. It returns boolean value.
 */
bool Periodic::compare_other(const string& other_string) const{

	if(description.find(other_string)!=string::npos){
		return true;
	}
	else if(series.find(other_string)!=string::npos){
		return true;
	}
	else if(Media::compare_other(other_string)){
		return true;
	}
	else if(related_titles.find(other_string)!=string::npos){
		return true;
	}
	else{
		return false;
	}
}

/*
   display function to print Periodics attributes. This function gives
   call to display() of Media class.
 */
void Periodic::display() const
{
	cout << "------Periodic------" << endl;
	Media::display();
	cout<<"Author:      	     "<<author<<endl;
	cout<<"Description:        "<<description<<endl;
	cout<<"Publisher:          "<<publisher<<endl;
	cout<<"publishing_history: "<<publishing_history<<endl;
	cout<<"Series: 	     "<<series<<endl;
	cout<<"related_titles:     "<<related_titles;
	cout<<"other_forms_of_title: "<<other_forms_of_title<<endl;
	cout<<"gov_doc_number:     "<<gov_doc_number<<endl<<endl;

}

File: libraryProject_Python/Periodic.py
from Media import Media

class Periodic(Media):

	def __init__(self,callNo,title,subject,author,description,publisher,publishing_history,series,notes,related_titles,other_forms_of_title,gov_doc_number):
		self.callNo = callNo
		self.title = title
		self.subject = subject
		self.author = author
		self.description = description
		self.publisher = publisher
		self.publishing_history = publishing_history
		self.series = series
		self.notes = notes
		self.related_titles = related_titles
		self.other_forms_of_title = other_forms_of_title
		self.gov_doc_number = gov_doc_number
		super(Periodic,self).__init__(self.callNo,self.title,self.subject,self.notes)

#	Media.__init__(self,callNo,title,subject,notes)
#	def __repr__(self):
#		return "%s %s %s %s %s %s %s %s %s %s" %#(self.callNo,self.title,self.subject,self.author,self.description,self.publisher,self.city,self.year,self.series,self.notes)


	
	def compare_Other(self,other):
		if other in self.description:
			return True
		elif other in self.series:
			return True
		elif other in self.related_titles:
			return True
		elif other in self.notes:
			return True
		else:
			return False




	def display(self):
		super(Periodic,self).display()
		print ("Author: %s" % (self.author))
		print ("Description: %s" % (self.description))
		print ("Publisher: %s" % (self.publisher))
		print ("publishing_history: %s" % (self.publishing_history))
		print ("Series: %s " % (self.series))
		print ("related_titles: %s" % (self.related_titles))
		print ("other_forms_of_title: %s" % (self.other_forms_of_title))
		print ("gov_doc_number: %s\n\n" % (self.gov_doc_number))

		
